src/mpi/coll/src/coll_impl.c

```diff
@@ -101,52 +147,141 @@ int MPIR_COLL_init(void) {
     /* Exscan Inter */
 
     /* Gather Intra */
+    if (0 == strcmp(MPIR_CVAR_GATHER_ALGORITHM_INTRA, "binomial"))
+        MPIR_Gather_alg_intra_choice = MPIR_GATHER_ALG_INTRA_BINOMIAL;
+    else
+        MPIR_Gather_alg_intra_choice = MPIR_GATHER_ALG_INTRA_AUTO;
 
     /* Gather Inter */
+    MPIR_Gather_alg_inter_choice = MPIR_GATHER_ALG_INTER_AUTO;
 
     /* Gatherv Intra */
```

Comment:
We're still missing a bunch of collectives in this list. Even if we only have the `_AUTO` options, I think we should still put the infrastructure in here to do the algorithm selection.

Reply:
@wesbland There are no inter and intra functions defined within gatherv and other functions. Is it ok to define them in coll_impl.c?

Reply:
Do the same things we do for the other functions, but then have them both call the same function.

Reply:
@wesbland Similarly allreduce_group has only intra and inter is not defined.

Reply:
Hmm. Is it illegal to do an allreduce_group on an intercommunicator? I'm not sure what the right answer is. @pavanbalaji?

Reply:
I think it should look the same as the others, string comparisons and all. You'd create the CVARs in `gather.c`, but there would only be one option (the same way we only have one option for most intercommunicator collectives).

Reply:
@wesbland For example, In Allgatherv Inter in coll_impl.c, As there is only one option we have not used string comparison. Should this also be changed?

Reply:
@wesbland allreduce_group is an internal function that is only used on intracommunicators.

Reply:
@wesbland  May I request you to give a pass on the following:
    /* Scatterv Intra */
    if (0 == strcmp(MPIR_CVAR_SCATTERV_ALGORITHM_INTRA, "linear"))
        MPIR_Scatterv_alg_intra_choice = MPIR_SCATTERV_ALG_LINEAR;
    else
        MPIR_Scatterv_alg_intra_choice = MPIR_SCATTERV_ALG_AUTO;

    /* Scatterv Inter */
    if (0 == strcmp(MPIR_CVAR_SCATTERV_ALGORITHM_INTER, "linear"))
        MPIR_Scatterv_alg_inter_choice = MPIR_SCATTERV_ALG_LINEAR;
    else
        MPIR_Scatterv_alg_inter_choice = MPIR_SCATTERV_ALG_AUTO;

Reply:
@pavanbalaji - Ok, then let's drop the intercommunicators here.

@Bayyapu - Yep, that looks right to me.

Reply:
For the collectives where intercommunicators are not allowed, we should throw an error instead of simply pointing to the same function (in a separate patch).